lib/subscription/batcher.ex

```diff
@@ -31,16 +31,20 @@ defmodule AshGraphql.Subscription.Batcher do
   end
 
   def publish(topic, notification, pubsub, key_strategy, doc) do
-    case GenServer.call(
-           __MODULE__,
-           {:publish, topic, notification, pubsub, key_strategy, doc},
-           :infinity
-         ) do
-      :handled ->
-        :ok
-
-      :backpressure_sync ->
-        do_send(topic, [notification], pubsub, key_strategy, doc)
+    if is_nil(Process.whereis(__MODULE__)) do
```

Comment:
I tried it with `catch` but that didn't seem to work.

Reply:
🤔 in what way? Ideally we wouldn't lookup the process twice (one to check and one to send) and then we also have the scenario where it's up when we check but crashes when we call it. The catch version *should* work.

Reply:
If what I read is correct, it is because it is not raising an error but sending an exit signal instead. 

This is what Claude came up with, but this felt even more complex to me

```elixir
defmodule SafeCall do
  def call(pid, message, timeout \\ 5000) do
    ref = Process.monitor(pid)
    send(pid, {:$gen_call, {self(), ref}, message})

    receive do
      {:DOWN, ^ref, :process, ^pid, reason} ->
        {:error, reason}
      {^ref, reply} ->
        Process.demonitor(ref, [:flush])
        {:ok, reply}
    after
      timeout ->
        Process.demonitor(ref, [:flush])
        {:error, :timeout}
    end
  end
end

# Usage
case SafeCall.call(pid, :some_message) do
  {:ok, reply} ->
    # Process the reply
  {:error, reason} ->
    # Handle the error (e.g., :noproc, :normal, :killed, etc.)
end
```

Reply:
I'm just talking about the `monitor`/`demonitor` stuff. Not the whole thing

Reply:
exits can be caught 🤔 

This works for me:

```elixir
try do
  GenServer.call(:not_existing, :hello)
catch
  :exit, {:noproc, _} -> "no process"
end
```

Reply:
exits from *linked processes* cannot be caught, but the exit from `exit/1` can be, which is what this is.

Reply:
Strange, I added it just like this. But the test still failed with the same error. I'll try again.

Reply:
Ok, my bad. I just tested with a "catch all" pattern like

```
catch
  err -> dbg(err0
```

to try and see exactly how the thing would look like, but in that case it wouldn't catch it.

Reply:
Oh, ok. because there are 2 params. `exit` and `{:noproc, _}` 

Is there a way to capture everything regardless of number of parameters. I thought that catch would always just have patterns with one parameter. 

TIL :sweat_smile:

Reply:
I think in this case the only thing we want to capture is the noproc, but you can just add
```elixir
catch
_ -> ...something was thrown
_, _ -> ...exit
```